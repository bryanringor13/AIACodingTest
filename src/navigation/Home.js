/* eslint-disable react/react-in-jsx-scope */
import React from 'react';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import Home from '../screens/home/Home';

const HomeStack = createNativeStackNavigator();

export const HomeNavigator = () => {
  return (
    <HomeStack.Navigator
      screenOptions={{
        headerShown: false,
      }}>
      <HomeStack.Screen name="Home" component={Home} />
    </HomeStack.Navigator>
  );
};
