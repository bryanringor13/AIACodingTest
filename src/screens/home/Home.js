import {SafeAreaView} from 'react-native';
import React from 'react';
import Assesment from './Assesment';
import {HomeProvider} from '../../hooks/home';
import Challenges from './Challenges';
import Logout from './Logout';

const Home = () => {
  return (
    <HomeProvider>
      <SafeAreaView>
        <Logout />
        <Assesment />
        <Challenges />
      </SafeAreaView>
    </HomeProvider>
  );
};

export default Home;
