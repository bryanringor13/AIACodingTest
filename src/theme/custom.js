export const currentTheme = {
  fontSize: {
    sm: '12px',
    md: '16px',
    lg: '18px',
    xl: '24px',
  },
  color: {
    danger: '#df4759',
    success: '#42ba96',
    link: '#0000EE',
  },
};
