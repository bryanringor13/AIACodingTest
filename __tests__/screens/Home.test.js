import React from 'react';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'styled-components/native';
import Home from '../../src/screens/home/Home';
import {currentTheme} from '../../src/theme/custom';

global.fetch = jest.fn(() => new Promise(resolve => resolve()));

describe('Home Screen', () => {
  it('renders <Home /> correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={currentTheme}>
          <Home />
        </ThemeProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
