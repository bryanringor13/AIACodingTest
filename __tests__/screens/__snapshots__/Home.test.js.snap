// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Screen renders <Home /> correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "justifyContent": "center",
        "marginRight": 10,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "color": "blue",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#0000EE",
              "fontColor": "#000000",
              "fontSize": 12,
            },
          ]
        }
        variant="link"
      >
        Logout
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginVertical": 10,
          "paddingLeft": 5,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "marginLeft": 5,
          },
        ]
      }
    >
      Assesments
    </Text>
    <View
      style={
        Array [
          Object {
            "width": "100%",
          },
        ]
      }
      width={525}
    >
      <RCTScrollView
        data={
          Array [
            Object {
              "id": "0001",
              "pts": 1000,
              "source": Object {
                "testUri": "../../../src/res/images/assesments/medicine_list.png",
              },
              "title": "Advanced Health Screening",
            },
            Object {
              "id": "0002",
              "pts": 50,
              "source": Object {
                "testUri": "../../../src/res/images/assesments/medicine_bottle.png",
              },
              "title": "Medicine Maintenance",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              dir="row"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "marginLeft": 5,
                    "marginRight": 10,
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                ]
              }
            >
              <View
                imageWidth={false}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../src/res/images/assesments/medicine_list.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "height": 100,
                        "width": 100,
                      },
                    ]
                  }
                />
              </View>
              <View
                dir="row"
                style={
                  Array [
                    Object {
                      "2625": "column",
                      "justifyContent": "space-between",
                      "marginLeft": 5,
                    },
                  ]
                }
                width={262.5}
              >
                <View>
                  <Text
                    dir="row"
                    style={
                      Array [
                        Object {
                          "fontSize": 16,
                          "fontWeight": "bold",
                          "marginBottom": 5,
                        },
                      ]
                    }
                  >
                    Advanced Health Screening
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "marginTop": 10,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "width": "60%",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "gray",
                          },
                        ]
                      }
                    >
                      Earn up to 
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "rgb(56, 110, 157)",
                              "fontSize": 16,
                              "fontWeight": "bold",
                            },
                          ]
                        }
                      >
                        1000
                      </Text>
                       pts
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-end",
                          "justifyContent": "flex-end",
                          "width": "40%",
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "rgb(123,32,76)",
                            "fontSize": 18,
                          },
                          undefined,
                          Object {
                            "fontFamily": "anticon",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      î˜›
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              dir="row"
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 5,
                    "flexDirection": "row",
                    "marginLeft": 5,
                    "marginRight": 10,
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                ]
              }
            >
              <View
                imageWidth={false}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../src/res/images/assesments/medicine_bottle.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "height": 100,
                        "width": 100,
                      },
                    ]
                  }
                />
              </View>
              <View
                dir="row"
                style={
                  Array [
                    Object {
                      "2625": "column",
                      "justifyContent": "space-between",
                      "marginLeft": 5,
                    },
                  ]
                }
                width={262.5}
              >
                <View>
                  <Text
                    dir="row"
                    style={
                      Array [
                        Object {
                          "fontSize": 16,
                          "fontWeight": "bold",
                          "marginBottom": 5,
                        },
                      ]
                    }
                  >
                    Medicine Maintenance
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "marginTop": 10,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "width": "60%",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "gray",
                          },
                        ]
                      }
                    >
                      Earn up to 
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "rgb(56, 110, 157)",
                              "fontSize": 16,
                              "fontWeight": "bold",
                            },
                          ]
                        }
                      >
                        50
                      </Text>
                       pts
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-end",
                          "justifyContent": "flex-end",
                          "width": "40%",
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "rgb(123,32,76)",
                            "fontSize": 18,
                          },
                          undefined,
                          Object {
                            "fontFamily": "anticon",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      î˜›
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "marginVertical": 10,
          "paddingLeft": 5,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "marginLeft": 5,
          },
        ]
      }
    >
      Challenges
    </Text>
    <View
      style={
        Array [
          Object {
            "width": "100%",
          },
        ]
      }
      width={525}
    >
      <RCTScrollView
        data={
          Array [
            Object {
              "id": "0001",
              "pts": 1000,
              "remaining": 6,
              "source": Object {
                "testUri": "../../../src/res/images/challenges/no_sugar.png",
              },
              "title": "Say no to sugar",
            },
            Object {
              "id": "0002",
              "pts": 50,
              "source": Object {
                "testUri": "../../../src/res/images/challenges/running.png",
              },
              "title": "5km challenge",
            },
          ]
        }
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        viewabilityConfigCallbackPairs={Array []}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 5,
                    "marginLeft": 5,
                    "marginRight": 10,
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                ]
              }
            >
              <View
                imageWidth={262.5}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "width": 262.5,
                    },
                  ]
                }
              >
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../src/res/images/challenges/no_sugar.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "height": 100,
                        "width": 100,
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "2625": "column",
                      "height": 200,
                      "justifyContent": "space-between",
                      "marginLeft": 5,
                    },
                  ]
                }
                width={262.5}
              >
                <View>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 16,
                          "fontWeight": "bold",
                          "marginBottom": 5,
                          "marginTop": 10,
                        },
                      ]
                    }
                  >
                    Say no to sugar
                  </Text>
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "gray",
                        },
                      ]
                    }
                  >
                    6
                     days left
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "marginTop": 10,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "width": "60%",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "gray",
                          },
                        ]
                      }
                    >
                      Earn up to 
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "rgb(56, 110, 157)",
                              "fontSize": 16,
                              "fontWeight": "bold",
                            },
                          ]
                        }
                      >
                        1000
                      </Text>
                       pts
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-end",
                          "justifyContent": "flex-end",
                          "width": "40%",
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "rgb(123,32,76)",
                            "fontSize": 18,
                          },
                          undefined,
                          Object {
                            "fontFamily": "anticon",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      î˜›
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 5,
                    "marginLeft": 5,
                    "marginRight": 10,
                    "marginTop": 10,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                ]
              }
            >
              <View
                imageWidth={262.5}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                      "width": 262.5,
                    },
                  ]
                }
              >
                <Image
                  resizeMode="contain"
                  source={
                    Object {
                      "testUri": "../../../src/res/images/challenges/running.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "height": 100,
                        "width": 100,
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "2625": "column",
                      "height": 200,
                      "justifyContent": "space-between",
                      "marginLeft": 5,
                    },
                  ]
                }
                width={262.5}
              >
                <View>
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 16,
                          "fontWeight": "bold",
                          "marginBottom": 5,
                          "marginTop": 10,
                        },
                      ]
                    }
                  >
                    5km challenge
                  </Text>
                </View>
                <View
                  style={
                    Array [
                      Object {
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                        "marginTop": 10,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "width": "60%",
                        },
                      ]
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "gray",
                          },
                        ]
                      }
                    >
                      Earn up to 
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "rgb(56, 110, 157)",
                              "fontSize": 16,
                              "fontWeight": "bold",
                            },
                          ]
                        }
                      >
                        50
                      </Text>
                       pts
                    </Text>
                  </View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "flex-end",
                          "justifyContent": "flex-end",
                          "width": "40%",
                        },
                      ]
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "rgb(123,32,76)",
                            "fontSize": 18,
                          },
                          undefined,
                          Object {
                            "fontFamily": "anticon",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      î˜›
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTSafeAreaView>
`;
